packages/firebase_auth/firebase_auth_web/lib/src/utils/web_utils.dart

```diff
@@ -14,6 +15,20 @@ import 'package:firebase_core_web/firebase_core_web_interop.dart'
 import '../interop/auth.dart' as auth_interop;
 import '../interop/multi_factor.dart' as multi_factor_interop;
 
+/// Workaround test to check whether `e` is practically a FirebaseError.
+///
+/// Ideally we'd check whether `e instanceof FirebaseError` however
+/// there are two definitions of `FirebaseError` in deployed apps, one from
+/// `firebase-auth.js` (which is usually minified) and one from
+/// `firebase-app.js`.
+///
+/// Because they are not the same, the instanceof check fails. Instead, we test
+/// that it is a window.Error (since that's the base class of FirebaseError) and
+/// we check that it defines "customData" property which is on the AuthError but not the FirebaseError it extends
```

Comment:
suggestion: I know these are all approximations, but given that the code below access `code` and `message` before it accesses `customData`, I wonder if it is safer to check for the code first, or check for both?

Another idea is to try to extract the code here, check that it starts with `auth/` as part of the validation, and return the code if it is available. For example:
```dart
String? _readFirebaseAuthErrorCode(Object o) {
  if (!instanceof(e, errorConstructor)) return null;
  String code = getProperty(e, 'code');
  if (code == null || !code.startsWith('auth/')) return null;
  return code.substring(5);
}

...

final code = _readFirebaseAuthErrorCode(exception);
if (code == null) {
  return FirebaseAuthException(code: 'unknown', ...);
}
...
```